public/entity2/entitysystem.h

```diff
@@ -0,0 +1,246 @@
+static_assert(sizeof(CEntitySystem) == 0x1510, "Class doesn't match expected size");
+
+// Size: 	0x1580 (from constructor)
+class CGameEntitySystem : public CEntitySystem
+{
+	struct SpawnGroupEntityFilterInfo_t
+	{
+		ISpawnGroupEntityFilter* m_pFilter;
+		SpawnGroupEntityFilterType_t m_nType;
+	};
+	//typedef SpawnGroupEntityFilterInfo_t CUtlMap<char const*, SpawnGroupEntityFilterInfo_t, int, bool (*)(char const* const&, char const* const&)>::ElemType_t;
+
+
+public:
+	virtual void		BuildResourceManifest(EntityResourceManifestCreationCallback_t callback, void* pContext, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest) override;
+	virtual void		BuildResourceManifest(const char* pManifestNameOrGroupName, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest) override;
+	virtual void		BuildResourceManifest(SpawnGroupHandle_t hSpawnGroup, const CUtlVector<const CEntityKeyValues*, CUtlMemory<const CEntityKeyValues*, int> >* pEntityKeyValues, const char* pFilterName, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest) override;
+	virtual void		BuildResourceManifest(SpawnGroupHandle_t hSpawnGroup, int nEntityKeyValueCount, const CEntityKeyValues** ppEntityKeyValues, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest) override;
+	virtual void		UnknownFunc004() override;
+	virtual void		BuildResourceManifestForEntity(uint64 unknown1, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest, uint64 unknown2) override;
+	virtual void		InvokePrecacheCallback(void* hResource, const EntitySpawnInfo_t* const info, IEntityPrecacheConfiguration* pConfig, IEntityResourceManifest* pResourceManifest, char* unk, void* callback) override;
+	virtual void		LockResourceManifest(bool bLock, CEntityResourceManifestLock* const context) override;
+	virtual				~CGameEntitySystem();
+	virtual void		ClearEntityDatabase(ClearEntityDatabaseMode_t eMode) override;
+	virtual void		FindEntityProcedural(const char* szName, CEntityInstance* pSearchingEntity, CEntityInstance* pActivator, CEntityInstance* pCaller) override;
+	virtual void		OnEntityParentChanged(CEntityInstance* pEntity, CEntityInstance* pNewParent) override;
+	virtual void		OnAddEntity(CEntityInstance* pEnt, CEntityHandle handle) override;
+	virtual void		OnRemoveEntity(CEntityInstance* pEnt, CEntityHandle handle) override;
+	virtual int			GetSpawnGroupWorldId(SpawnGroupHandle_t hSpawnGroup) override;
+	virtual void		Spawn(int nCount, const EntitySpawnInfo_t* pInfo) override;
+	virtual void		Activate(int nCount, const EntityActivation_t* pActivates, ActivateType_t activateType) override;
+	virtual void		PostDataUpdate(int nCount, const PostDataUpdateInfo_t* pInfo) override;
+	virtual void		OnSetDormant(int nCount, const EntityDormancyChange_t* pInfo, bool bNotifyAddedToPVS) override;
+	virtual void		UpdateOnRemove(int nCount, const EntityDeletion_t* pDeletion) override;
```

Comment:
Not sure if these needs to be defined again with an override flag? Since we aren't going to implement them ourselves this could be removed as it'll inherit these from the inherited class anyway.

Reply:
I'm not sure how class declaration work in C++ so I was afraid that the offsets will be incorrect if I don't do that. I removed the overrides now, feel free to let me know if it's still wrong

Reply:
The virtual functions gets inherited from the base class, so functions from ``BuildResourceManifest`` to ``LockResourceManifest`` could be removed from here as they are defined in the parent class, and others should be leaved as is but without the override flag, just as a pure virtuals (= 0;)